### PR TITLE
Allow access to the Public Organization Member lists with minimal permissions

### LINKED ISSUE #7501: Display org members if org is public
Right now, you can't view the org members page even if an organization is public:

 https://gitea.com/org/gitea/members

That will ask for a sign in. Even once signed in, that page shows a 404 if you are not a member of the org. Public org member lists should be viewable by anybody (with individual users hidden if they choose). This is needed for being able to update README files when migrating projects to gitea.com

### PR DESCRIPTION
**Pull request by a1012112796**
Examining Organization membership should not necessarily require sign-in if the organization is public and the members are public. Therefore we should adjust `/org/{org}/members` to not require login.

Fix #7501

Signed-off-by: a1012112796 <1012112796@qq.com>

### CODE REVIEW

**Review by Gusted — APPROVED** [review]

**Review by wxiaoguang — DISMISSED** [review]

File: routers/web/web.go [hunk 1]
```diff
@@ -610,6 +610,12 @@ func RegisterRoutes(m *web.Route) {
 	}
 
 	// ***** START: Organization *****
+	m.Group("/org", func() {
+		m.Group("/{org}", func() {
+			m.Get("/members", org.Members)
+		}, context.OrgAssignment())
```
Comment (zeripath) [inline routers/web/web.go:616]: OrgAssignment and HandleOrgAssignment don't appear to be correctly handling visibility in here.

**Review by zeripath — CHANGES_REQUESTED** [review]
This PR reveals the existence of private organisations as it stands.

**Comment by zeripath** [comment]
```patch
diff --git a/modules/context/org.go b/modules/context/org.go
index 9f4ce485e..fd808403d 100644
--- a/modules/context/org.go
+++ b/modules/context/org.go
@@ -12,6 +12,7 @@ import (
 	"code.gitea.io/gitea/models/perm"
 	user_model "code.gitea.io/gitea/models/user"
 	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/structs"
 )
 
 // Organization contains organization context
@@ -69,6 +70,18 @@ func HandleOrgAssignment(ctx *Context, args ...bool) {
 		return
 	}
 	org := ctx.Org.Organization
+
+	// Handle Visibility
+	if org.Visibility != structs.VisibleTypePublic && !ctx.IsSigned {
+		// We must be signed in to see limited or private organizations
+		ctx.NotFound("OrgAssignment", err)
+		return
+	} else if org.Visibility == structs.VisibleTypePrivate {
+		requireMember = true
+	} else if ctx.IsSigned && ctx.Doer.IsRestricted {
+		requireMember = true
+	}
+
 	ctx.ContextUser = org.AsUser()
 	ctx.Data["Org"] = org
 
diff --git a/routers/web/org/home.go b/routers/web/org/home.go
index d565a0c24..63243a391 100644
--- a/routers/web/org/home.go
+++ b/routers/web/org/home.go
@@ -39,11 +39,6 @@ func Home(ctx *context.Context) {
 
 	org := ctx.Org.Organization
 
-	if !organization.HasOrgOrUserVisible(ctx, org.AsUser(), ctx.Doer) {
-		ctx.NotFound("HasOrgOrUserVisible", nil)
-		return
-	}
-
 	ctx.Data["PageIsUserProfile"] = true
 	ctx.Data["Title"] = org.DisplayName()
 	if len(org.Description) != 0 {

```

I think this is required.

File: routers/web/web.go (hunk 1 above)
Comment (wxiaoguang) [reply routers/web/web.go:616]: Good catch, I only checked that there is `IsOrgMember` in `org.Members` before.

Maybe it's worth to add a test for it to make sure the visibility handling is correct.

**Review by zeripath — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Comment by zeripath** [comment]
Make lgtm work

**Pull request merged**